src/StreamJsonRpc.Tests/JsonRpcClientInteropTests.cs

```diff
@@ -44,6 +41,82 @@ public JsonRpcClientInteropTests(ITestOutputHelper logger)
     }
 
     [Fact]
+    public async Task NotifyAsync_ParameterObjectSentAsArray()
+    {
+        Task notifyTask = this.clientRpc.NotifyAsync("test", new  { Bar = "test" });
+        JObject request = await this.ReceiveAsync();
+        Assert.Equal(JTokenType.Array, request["params"].Type);
+        Assert.Equal("test", ((JArray)request["params"])[0]["Bar"].ToString());
+    }
+
+    [Fact]
+    public async Task NotifyAsync_NoParameter()
+    {
+        Task notifyTask = this.clientRpc.NotifyAsync("test");
+        JObject request = await this.ReceiveAsync();
+        Assert.Equal(JTokenType.Array, request["params"].Type);
+        Assert.Equal(0, ((JArray)request["params"]).Count);
+    }
+
+    [Fact]
+    public async Task NotifyWithParameterPassedAsObjectAsync_ParameterObjectSentAsObject()
+    {
+        Task notifyTask = this.clientRpc.NotifyWithParameterObjectAsync("test", new { Bar = "test" });
+        JObject request = await this.ReceiveAsync();
+        Assert.Equal(JTokenType.Object, request["params"].Type);
+        Assert.Equal("test", request["params"]["Bar"].ToString());
+    }
+
+    [Fact]
+    public async Task NotifyWithParameterPassedAsObjectAsync_NoParameter()
+    {
+        Task notifyTask = this.clientRpc.NotifyWithParameterObjectAsync("test");
+        JObject request = await this.ReceiveAsync();
+        Assert.Null(request["params"]);
+    }
+
+    [Fact]
+    public async Task InvokeAsync_ParameterObjectSentAsArray()
+    {
+        Task notifyTask = this.clientRpc.InvokeAsync<object>("test", new { Bar = "test" });
+        JObject request = await this.ReceiveAsync();
+        Assert.Equal(JTokenType.Array, request["params"].Type);
+        Assert.Equal("test", ((JArray)request["params"])[0]["Bar"].ToString());
+    }
+
+    [Fact]
+    public async Task InvokeAsync_NoParameter()
+    {
+        Task notifyTask = this.clientRpc.InvokeAsync<object>("test");
+        JObject request = await this.ReceiveAsync();
+        Assert.Equal(JTokenType.Array, request["params"].Type);
+        Assert.Equal(0, ((JArray)request["params"]).Count);
+    }
+
+    [Fact]
+    public async Task InvokeWithParameterPassedAsObjectAsync_ParameterObjectSentAsObject()
+    {
+        Task notifyTask = this.clientRpc.InvokeWithParameterObjectAsync<object>("test", new { Bar = "test" });
```

Comment:
tip for next time: use different values for the various strings so that you detect bugs in your assertions where the values get mixed up in the product. e.g. if the method name and parameter got mixed up, they'd still both be "test" and your test might pass.